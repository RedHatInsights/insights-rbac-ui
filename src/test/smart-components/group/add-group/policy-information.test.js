import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import PolicyInformation from '../../../../smart-components/group/add-group/policy-information';

describe('<PolicyInformation />', () => {
  let initialProps;

  beforeEach(() => {
    initialProps = {
      title: 'Policy info title',
      editType: 'add',
      formData: {
        policy: {
          name: 'Test policy name',
          description: 'Test policy description',
        },
      },
      onHandleChange: jest.fn(),
      setIsPolicyInfoValid: jest.fn(),
    };
  });

  it('should render correctly', () => {
    const wrapper = mount(<PolicyInformation {...initialProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
