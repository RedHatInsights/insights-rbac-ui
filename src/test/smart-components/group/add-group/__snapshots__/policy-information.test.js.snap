// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PolicyInformation /> should render correctly 1`] = `
<PolicyInformation
  editType="add"
  formData={
    Object {
      "policy": Object {
        "description": "Test policy description",
        "name": "Test policy name",
      },
    }
  }
  onHandleChange={[MockFunction]}
  setIsPolicyInfoValid={[MockFunction]}
  title="Policy info title"
>
  <Form>
    <form
      className="pf-c-form"
      noValidate={true}
    >
      <Stack
        hasGutter={true}
      >
        <div
          className="pf-l-stack pf-m-gutter"
        >
          <StackItem>
            <div
              className="pf-l-stack__item"
            >
              <PolicyInfoText
                editType="add"
                title="Policy info title"
              >
                <StackItem>
                  <div
                    className="pf-l-stack__item"
                  >
                    <Title
                      headingLevel="h4"
                      size="xl"
                    >
                      <h4
                        className="pf-c-title pf-m-xl"
                      >
                        Policy info title
                      </h4>
                    </Title>
                  </div>
                </StackItem>
                <TextContent>
                  <div
                    className="pf-c-content"
                  >
                    <Text
                      component="h6"
                    >
                      <h6
                        className=""
                        data-pf-content={true}
                      >
                        Policies are the permissions set for this group. Groups can have one or more policies. Policies are created for a group, they cannot be shared. You can only create one policy at this time. It is possible to create more for this group at a later time.
                        <br />
                        All fields are optional.
                      </h6>
                    </Text>
                  </div>
                </TextContent>
              </PolicyInfoText>
            </div>
          </StackItem>
          <StackItem>
            <div
              className="pf-l-stack__item"
            >
              <FormGroup
                fieldId="policy-name"
                isValid={true}
                label="Name"
              >
                <div
                  className="pf-c-form__group"
                  isValid={true}
                >
                  <div
                    className="pf-c-form__group-label"
                  >
                    <label
                      className="pf-c-form__label"
                      htmlFor="policy-name"
                    >
                      <span
                        className="pf-c-form__label-text"
                      >
                        Name
                      </span>
                    </label>
                     
                  </div>
                  <div
                    className="pf-c-form__group-control"
                  >
                    <ForwardRef
                      aria-describedby="policy-name"
                      id="policy-name"
                      name="policy-name"
                      onBlur={[Function]}
                      onChange={[Function]}
                      type="text"
                      value="Test policy name"
                    >
                      <TextInputBase
                        aria-describedby="policy-name"
                        aria-label={null}
                        className=""
                        id="policy-name"
                        innerRef={null}
                        isDisabled={false}
                        isLeftTruncated={false}
                        isReadOnly={false}
                        isRequired={false}
                        name="policy-name"
                        onBlur={[Function]}
                        onChange={[Function]}
                        type="text"
                        validated="default"
                        value="Test policy name"
                      >
                        <input
                          aria-describedby="policy-name"
                          aria-invalid={false}
                          aria-label={null}
                          className="pf-c-form-control"
                          disabled={false}
                          id="policy-name"
                          name="policy-name"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          readOnly={false}
                          required={false}
                          type="text"
                          value="Test policy name"
                        />
                      </TextInputBase>
                    </ForwardRef>
                  </div>
                </div>
              </FormGroup>
            </div>
          </StackItem>
          <StackItem>
            <div
              className="pf-l-stack__item"
            >
              <FormGroup
                fieldId="policy-description"
                label="Description"
              >
                <div
                  className="pf-c-form__group"
                >
                  <div
                    className="pf-c-form__group-label"
                  >
                    <label
                      className="pf-c-form__label"
                      htmlFor="policy-description"
                    >
                      <span
                        className="pf-c-form__label-text"
                      >
                        Description
                      </span>
                    </label>
                     
                  </div>
                  <div
                    className="pf-c-form__group-control"
                  >
                    <TextArea
                      aria-label={null}
                      className=""
                      id="policy-description"
                      isRequired={false}
                      name="policy-description"
                      onChange={[Function]}
                      resizeOrientation="both"
                      type="text"
                      validated="default"
                      value="Test policy description"
                    >
                      <textarea
                        aria-invalid={false}
                        aria-label={null}
                        className="pf-c-form-control"
                        id="policy-description"
                        name="policy-description"
                        onChange={[Function]}
                        required={false}
                        type="text"
                        value="Test policy description"
                      />
                    </TextArea>
                  </div>
                </div>
              </FormGroup>
            </div>
          </StackItem>
        </div>
      </Stack>
    </form>
  </Form>
</PolicyInformation>
`;
